modules/db/src/main/scala/gem/dao/UserDao.scala

```diff
@@ -13,29 +13,10 @@ object UserDao {
     select("root")
 
   def select(id: String): ConnectionIO[User[Nothing]] =
-    sql"""
-      SELECT id, first, last, email, staff
-      FROM gem_user
-      WHERE id = $id
-    """.query[(String, String, String, String, Boolean)]
-       .unique.map { case (i, f, l, e, s) =>
-         User(i, f, l, e, s, Map.empty)
-       }
+    Statements.select(id).unique
```

Comment:
Yeah, checking needs to see a more precise structure than `ConnectionIO`.

`sql" ... $a ... $b ... ".query[Foo]` gives you a `Query0` which has enough information to perform a metadata check. That's what lives in the `Statements` modules and what is used by the tests.

When you then say `.list` you have a `ConnectionIO` which no longer knows it's a query; it's just a program that produces a `List[Foo]`. This is what you care about as a user and what you want to expose on the DAO.

Analogous for updates.